src/actions.ts

```diff
@@ -915,4 +1011,212 @@ export function createActionsSystem(
+      const parent = getEntityParent(entity)
+      return !parent ? entity : getRoot(parent)
+    }
+
+    for (const target of damageTargets) {
+      const targetPosition = getWorldPosition(target)
+      const distance = Vector3.distance(entityPosition, targetPosition)
+
+      if (layer) {
+        if (layer === ProximityLayer.PLAYER) {
+          const root = getRoot(target)
+          if (root !== engine.PlayerEntity && root !== engine.CameraEntity) {
+            continue
+          }
+        } else if (layer === ProximityLayer.NON_PLAYER) {
+          const root = getRoot(target)
+          if (root === engine.PlayerEntity || root === engine.CameraEntity) {
+            continue
+          }
+        }
+      }
+      if (distance <= radius) {
+        const total = hits === undefined ? 1 : Math.max(hits, 1)
+        for (let i = 0; i < total; i++) {
+          const triggerEvents = getTriggerEvents(target)
+          triggerEvents.emit(TriggerType.ON_DAMAGE)
+        }
+      }
+    }
+  }
+
+  // MOVE_PLAYER_HERE
+  function handleMovePlayerHere(
+    entity: Entity,
+    _payload: ActionPayload<ActionType.MOVE_PLAYER_HERE>,
+  ) {
+    const here = getWorldPosition(entity)
+    void movePlayerTo({ newRelativePosition: here })
+  }
+
+  // PLACE_ON_PLAYER
+  function handlePlaceOnPlayer(
+    entity: Entity,
+    _payload: ActionPayload<ActionType.PLACE_ON_PLAYER>,
+  ) {
+    const transform = Transform.getMutableOrNull(entity)
+    const player = Transform.getOrNull(engine.PlayerEntity)
+    if (transform && player) {
+      transform.position = player.position
+    }
+  }
+
+  // ROTATE_AS_PLAYER
+  function handleRotateAsPlayer(
+    entity: Entity,
+    _payload: ActionPayload<ActionType.ROTATE_AS_PLAYER>,
+  ) {
+    const transform = Transform.getMutableOrNull(entity)
+    const player = Transform.getOrNull(engine.PlayerEntity)
+    if (transform && player) {
+      transform.rotation = player.rotation
+    }
+  }
+
+  // PLACE_ON_CAMERA
+  function handlePlaceOnCamera(
+    entity: Entity,
+    _payload: ActionPayload<ActionType.PLACE_ON_CAMERA>,
+  ) {
+    const transform = Transform.getMutableOrNull(entity)
+    const camera = Transform.getOrNull(engine.CameraEntity)
+    if (transform && camera) {
+      transform.position = camera.position
+    }
+  }
+
+  // ROTATE_AS_CAMERA
+  function handleRotateAsCamera(
+    entity: Entity,
+    _payload: ActionPayload<ActionType.ROTATE_AS_CAMERA>,
+  ) {
+    const transform = Transform.getMutableOrNull(entity)
+    const camera = Transform.getOrNull(engine.CameraEntity)
+    if (transform && camera) {
+      transform.rotation = camera.rotation
+    }
+  }
+
+  // SET_POSITION
+  function handleSetPosition(
+    entity: Entity,
+    payload: ActionPayload<ActionType.SET_POSITION>,
+  ) {
+    const transform = Transform.getMutableOrNull(entity)
+    if (transform) {
+      transform.position = Vector3.create(payload.x, payload.y, payload.z)
```

Comment:
should this position be relative to the scene, or is it okay to be global?

```
transform.position = Vector3.create(
  transform.position.x + payload.x,
  transform.position.y + payload.y,
  transform.position.z + payload.z
)
```

Reply:
mm i can see use cases for both, although you can do the relative use case by nesting a child entity and using the SET_POSITION action on the child. But I think I could add a `relative?: boolean` param and if present add the values to the current ones as in your example

Reply:
done ✅